terraform/lib/dependabot/terraform/file_updater.rb

```diff
@@ -73,27 +87,86 @@ def updated_terraform_file_content(file)
         content
       end
 
-      def update_git_declaration(new_req, old_req, updated_content, filename)
+      def update_git_declaration_tf(new_req, old_req, updated_content, filename)
         url = old_req.fetch(:source)[:url].gsub(%r{^https://}, "")
         tag = old_req.fetch(:source)[:ref]
         url_regex = /#{Regexp.quote(url)}.*ref=#{Regexp.quote(tag)}/
 
         declaration_regex = git_declaration_regex(filename)
 
-        updated_content.sub!(declaration_regex) do |regex_match|
+        updated_content.sub(declaration_regex) do |regex_match|
           regex_match.sub(url_regex) do |url_match|
             url_match.sub(old_req[:source][:ref], new_req[:source][:ref])
           end
         end
       end
 
-      def update_registry_declaration(new_req, old_req, updated_content)
+      def update_git_declaration_json(new_req, old_req, updated_content)
+        url = old_req.fetch(:source)[:url].gsub(%r{^https://}, "")
+        tag = old_req.fetch(:source)[:ref]
+        url_regex = /#{Regexp.quote(url)}.*ref=#{Regexp.quote(tag)}/
+
+        module_name = dependency.name.split("::").first
+        tf = JSON.parse(updated_content)
+        tf_module = tf["module"][module_name] || []
+        tf_module.each do |obj|
+          obj["source"].sub!(url_regex) do |url_match|
+            url_match.sub(old_req[:source][:ref], new_req[:source][:ref])
+          end
+        end
+        JSON.pretty_generate(tf)
```

Comment:
As you mentioned in your PR comment this may reformat the json which I think we'd prefer to avoid to limit noisy PRs at the very least.  I'm not a regex expert, but I think we should take a shot at building a regex to match/sub for json.  This would be the final piece to prevent branching as we could just handle this new regex the same as we currently do for terragrunt/terraform in `git_declaration_regex`

Reply:
Thanks for the feedback! I think you got to the bottom of it. Really the only reason for the distinction between .tf and .tf.json files in the fetcher was because I foresaw having to make the distinction in the updater. I didn't initially think about the messy diffs that this approach could create -- I could give the json regex a shot.